File: exercise/kotlin/day12/src/test/kotlin/GreeterTests.kt
import greeting.Greeter
import io.kotest.core.spec.style.StringSpec
import io.kotest.matchers.shouldBe

class GreeterTests : StringSpec({
    "say hello" {
        val greeter = Greeter()
        greeter.greet() shouldBe "Hello."
    }

    "say hello formally" {
        val greeter = Greeter("formal")
        greeter.greet() shouldBe "Good evening, sir."
    }

    "say hello casually" {
        val greeter = Greeter("casual")
        greeter.greet() shouldBe "Sup bro?"
    }

    "say hello intimately" {
        val greeter = Greeter("intimate")
        greeter.greet() shouldBe "Hello Darling!"
    }
})

File: solution/c#/Day12/Day12.Tests/GreeterTests.cs
using FluentAssertions;
using Xunit;
using static Day12.GreeterFactory;

namespace Day12.Tests
{
    public class GreeterTests
    {
        [Theory]
        [InlineData(GreeterFactory.Casual, "Sup bro?")]
        [InlineData(GreeterFactory.Formal, "Good evening, sir.")]
        [InlineData(GreeterFactory.Intimate, "Hello Darling!")]
        [InlineData("Random string", "Hello.")]
        public void Greet(string greeter, string expectedGreeter)
            => Create(greeter)
                .Greet()
                .Should().Be(expectedGreeter);
    }
}